### PR TITLE
Fix PERMISSIVE_RESIZE map

### PR DESCRIPTION
**Pull request by naoyam**
It was not a superset of the PERMISSIVE map.

This fix also exposed other issues in domain mappings and analyses, which are also fixed

Confirmed no benchmark is affected by this change.

### CODE REVIEW

**Comment by naoyam** [comment]
!build

File: test/test_resize.cpp [hunk 1]
```diff
@@ -1149,6 +1149,16 @@ TEST_F(NVFuserTest, FusionResizeSliceScheduler1_CUDA) {
         sub(tv0->axis(0)->extent(), IrBuilder::create<Scalar>(1L))}});
   fusion.addOutput(tv1);
 
+  // Make sure all IDs of tv0 and tv1 are mapped in the
+  // PERMISSIVE_RESIZE mode.
+  ComputeAtMap ca_map(&fusion);
+  ASSERT_TRUE(ca_map.areMapped(
+      tv1->axis(0), tv0->axis(0), IdMappingMode::PERMISSIVE_RESIZE));
+  ASSERT_TRUE(ca_map.areMapped(
```
Comment (naoyam) [inline test/test_resize.cpp:1157]: This is not true with the current main

File: csrc/scheduler/utils.cpp [hunk 2]
```diff
@@ -1089,7 +1090,12 @@ IterDomain* projectIdToRoot(
     } else if (expr->isA<Resize>()) {
       auto resize = expr->as<Resize>();
       if (resize->out() == projected_id) {
-        projected_id = nullptr;
+        // We do not allow vectorization wit resize at this moment
```
Comment (zasdfgbnm) [inline csrc/scheduler/utils.cpp:1093]: nit:
```suggestion
        // We do not allow vectorization with resize at this moment
```

**Review by zasdfgbnm — APPROVED** [review]
Thanks for fixing!

**Pull request merged**